src/main/java/edu/cornell/kfs/sys/service/impl/ReportWriterTextServiceImpl.java

```diff
@@ -66,6 +75,91 @@ public void initialize() {
+	    } //else - use spring configured default report file name
+	    initialize();
+	}
+	
+	/**
+	 * This method generates a file name prefix from the fully qualified file name attribute
+	 *     (i.e.   directoryPath/fileName.fileExtension)
+	 * This functionality was created for the CreateAccountingDocumentReportItem xmlFileName attribute but could
+	 * be utilized by any other batch job that desired the report output file contain the input data file name
+	 * as the report file prefix. The input parameter is assumed to contain the fully qualified directory path
+	 * as well as a file name and extension. This method strips off both the directory path and file extension
+	 * returning just the file name portion of the string.
+	 * When both a directory path and file extension are not detected, a zero length string is returned.
+	 *
+	 * Example:
+	 *  Input parameter : /infra/work/staging/fp/accountingXmlDocument/fp_ib_netsuite_20240229_050035.xml
+	 *  Return value    : fp_ib_netsuite_20240229_050035
+	 *  
+	 *  @param fullyQualifiedDataFileName
+	 */
+	private String obtainDataFileName(String fullyQualifiedDataFileName) {
+	    String onlyDataFileName = KFSConstants.EMPTY_STRING;
+	    
+	    if (StringUtils.contains(fullyQualifiedDataFileName, KFSConstants.DELIMITER)
+	            && (StringUtils.contains(fullyQualifiedDataFileName, File.separator))) {
+	        onlyDataFileName = removeFileExtension(fullyQualifiedDataFileName);
+	        
+	        if (StringUtils.isNotBlank(onlyDataFileName)) {
+	            onlyDataFileName = StringUtils.substringAfterLast(onlyDataFileName, File.separator);
+	        }
+	    }
+	    return onlyDataFileName;
+	}
+	
+	private String removeFileExtension(String fullyQualifiedDataFileName) {
+	    String filePathAndNameWithExtensionRemoved = KFSConstants.EMPTY_STRING;
+	    String [] subStringArray = StringUtils.split(fullyQualifiedDataFileName, KFSConstants.DELIMITER);
+	    if (ObjectUtils.isNotNull(subStringArray) 
+	            && subStringArray.length == 2) {
+	        filePathAndNameWithExtensionRemoved =  subStringArray[0];
+	    }
+	    return filePathAndNameWithExtensionRemoved;
+	}
+	
+	private boolean fullFilePathLimitWouldNotBeExceeded(String dataFileNamePrefix) {
+	    int numCharsCalculated = StringUtils.length(filePath) + File.separator.length() 
+	            + StringUtils.length(dataFileNamePrefix) + StringUtils.length(REPORT_FILE_NAME_INFIX) 
+	                + StringUtils.length(dateTimeService.toDateTimeStringForFilename(dateTimeService.getCurrentDate())) 
+	                    + StringUtils.length(fileNameSuffix);
+
+	    if (numCharsCalculated < NUM_CHARS_ALLOWED_FOR_FULL_FILE_PATH) {
+	        LOG.debug("fullFilePathLimitWouldNotBeExceeded: report file name prefix {} ok for path limit.", dataFileNamePrefix);
+	        return true;
+	    } else {
+	        LOG.warn("fullFilePathLimitWouldNotBeExceeded: Using data file XML name as prefix for report file prefix "
```

Comment:
Since the new feature has been updated to allow for more than just XML files, please update this message to avoid mentioning XML.

Reply:
Fixed.